tests/dragonfly/tls_replication_test.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+import asyncio
+import redis
+from redis import asyncio as aioredis
+from . import DflyInstanceFactory, dfly_args
+import logging
+import time
+
+BASE_PORT = 1111
+
+"""
+Test replication with tls and non-tls options
+"""
+
+# 1. Number of master threads
+# 2. Number of threads for each replica
+replication_cases = [(8, 8)]
+
+def tls_args(df_instance):
+    with_tls_args = {"tls": "",
+                    "tls_key_file": df_instance.dfly_path + "dfly-key.pem",
+                    "tls_cert_file": df_instance.dfly_path + "dfly-cert.crt",
+                    "no_tls_on_replica": "true"}
+    return with_tls_args
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("t_master, t_replica", replication_cases)
+@dfly_args({"dbfilename": "test-tls-replication-{{timestamp}}"})
+async def test_replication_all(df_local_factory, df_seeder_factory, t_master, t_replica):
+
+    # 1. Spin up dragonfly without tls, debug populate and then shut it down
+    master = df_local_factory.create(port=BASE_PORT, proactor_threads=t_master)
+    master.start()
+    c_master = aioredis.Redis(port=master.port)
+    await c_master.execute_command("DEBUG POPULATE 100")
+    await c_master.execute_command("SAVE")
+    db_size = await c_master.execute_command("DBSIZE")
+    assert 100 == db_size
+
+    master.stop()
+    with_tls_args = tls_args(master)
+    # 2. Spin up dragonfly again, this time, with different options:
+    #   a. Enable tls
+    #   b. Allow non-tls connection from replica to master
+    master = df_local_factory.create(**with_tls_args, port=BASE_PORT, proactor_threads=t_master)
+    master.start()
+    # we need to sleep, because currently tls connections on master fail
+    # TODO Fix this once master properly loads tls certificates and redis-cli does not
+    # require --insecure flag, therefore for now we can't use `wait_available`
+    time.sleep(10)
```

Comment:
Not a big fun of this, it should take a couple of `ms` to load the snapshot so I don't expect the code below to fail. 

This will be fixed once: 
1. Server works properly with TLS && redis-cli connects without the --insecure flag